sdcm/stress_thread.py

```diff
@@ -136,10 +136,27 @@ def create_stress_cmd(self, node, loader_idx, keyspace_idx):
 
     @staticmethod
```

Comment:
A question outside the review: why do we like static methods so much? This function does not access any "class global" fields (and rightly so), it is a pure function. Why not simply make it a function (outside class)?

Reply:
Reason in this particular case:
1. This function has no meaning outside of the class context.

IMHO, most significant plus that static methods gives you is ability, in case if it is needed, to change this part of logic in child class, in case of function you will have to duplicate all methods where you are using it, which is less efficient. Also moving code into standalone function shifts you from OOP toward functional programming, which is not how this product meant to be.

Reply:
> This function has no meaning outside of the class context.

It does. In fact, I don't  think it should be inside this class: the class has too many responsibilities already. The function can be reused in other contexts (e.g. if I want to run cassandra stress without creating a separate thread).

> ability, in case if it is needed, to change this part of logic in child class,

You ain't gonna need it...
And even if you do, that would make the code unreadable IMO.

> Also moving code into standalone function shifts you from OOP toward functional programming, which is not how this product meant to be.

OOP and functional programming are not incompatible.
Specifically, putting stuff into classes is not what defines OOP (IMO).

Also, sticking to one paradigm "because we want OOP" is not reasonable, it's cargo culting.

Reply:
> you will have to duplicate all methods where you are using it, which is less efficient.

What do you mean? You can pass the function as a parameter if you need to.

Reply:
> > This function has no meaning outside of the class context.
> 
> It does. In fact, I don't think it should be inside this class: the class has too many responsibilities already. The function can be reused in other contexts (e.g. if I want to run cassandra stress without creating a separate thread).

I agree on that, it has to be splitted into at least four classes.

> > ability, in case if it is needed, to change this part of logic in child class,

> You ain't gonna need it...
> And even if you do, that would make the code unreadable IMO.

You may need that with same probability as you may need to run c-s in the same thread. 
> > Also moving code into standalone function shifts you from OOP toward functional programming, which is not how this product meant to be.
> 
> OOP and functional programming are not incompatible.
> Specifically, putting stuff into classes is not what defines OOP (IMO).
> 
> Also, sticking to one paradigm "because we want OOP" is not reasonable, it's cargo culting.

You are not correct, you have to stick to paradigm that is used in the source base you are working on.

Reply:
> You may need that with same probability as you may need to run c-s in the same thread.

Perhaps. That was just a counterargument to the "meaning outside class context" argument. I agree that the counterargument is not much better than the argument itself.

Given that both probabilities are miniscule, we should use the simplest abstraction that is right for our current purposes. And the right abstraction here is a function - not a static method.

> You are not correct, you have to stick to paradigm that is used in the source base you are working on.

Why? I think we should use the right tool for the job, not stick to "one paradigm". Is there a particular reason SCT has to be "pure OOP", whatever "pure OOP" means?

Also, a "pure OOP" way would to be to have a singleton, not use static methods. But singletons are today understood as antipatterns, so we've got a problem.

Reply:
> > You may need that with same probability as you may need to run c-s in the same thread.
> 
> Perhaps. That was just a counterargument to the "meaning outside class context" argument. I agree that the counterargument is not much better than the argument itself.
> 
> Given that both probabilities are miniscule, we should use the simplest abstraction that is right for our current purposes. And the right abstraction here is a function - not a static method.

Not correct, probabilities are the same, but this source base uses OOP, and you have an object that is only user of this code now and near future, so you go staticmethod.

> > You are not correct, you have to stick to paradigm that is used in the source base you are working on.
> 
> Why? I think we should use the right tool for the job, not stick to "one paradigm". Is there a particular reason SCT has to be "pure OOP", whatever "pure OOP" means?

I'am not saying "pure", in this particulare case I can stick to it and I do it, when I don't - I don't.

> Also, a "pure OOP" way would to be to have a singleton, not use static methods. But singletons are today understood as an antipattern, so we've got a problem :P

Let's stick to the PR, if you want to chat find me in hangouts.

Reply:
The PR itself LGTM.